components/Map/DynamicMap.js

```diff
@@ -0,0 +1,123 @@
+  // center of map, right now coordinates of london
+  let center = [51.505, -0.09];
+
+  const [plan, setPlan] = useState({});
+  const [isValidPlan, setValidPlan] = useState(false);
+  const [positionFrom, setPositionFrom] = useState(null);
+  const [positionTo, setPositionTo] = useState(null);
+
+  // definition of start and stop marker image/size
+  const fromIcon = L.icon({
+    iconUrl: "/marker_pin.png",
+    iconRetinaUrl: "/marker_pin.png",
+    //shadowUrl: '/marker_pin.png',
+    //shadowRetinaUrl: './marker_pin.png',
+    iconSize: [24, 32], // size of the icon
+    //shadowSize:   [50, 64], // size of the shadow
+    iconAnchor: [12, 24], // point of the icon which will correspond to marker's location
+    //shadowAnchor: [4, 62],  // the same for the shadow
+    //popupAnchor:  [-3, -76] // point from which the popup should open relative to the iconAnchor
+  });
+
+  const toIcon = L.icon({
+    iconUrl: "/marker_pin.png",
+    iconRetinaUrl: "/marker_pin.png",
+    //shadowUrl: '/marker_pin.png',
+    //shadowRetinaUrl: './marker_pin.png',
+    iconSize: [24, 32], // size of the icon
+    //shadowSize:   [50, 64], // size of the shadow
+    iconAnchor: [12, 24], // point of the icon which will correspond to marker's location
+    //shadowAnchor: [4, 62],  // the same for the shadow
+    //popupAnchor:  [-3, -76] // point from which the popup should open relative to the iconAnchor
+  });
+
+  // fetch data from OTP server
+  useEffect(() => {
+    const fetchItems = async () => {
+      const result = await axios(
+        `${server}?fromPlace=${fromPlace}&toPlace=${toPlace}&mode=WALK&showIntermediateStops=true&maxStopToShapeSnapDistance=1`
+      );
+      setPlan(result.data.plan);
+    };
+    fetchItems();
+  }, []);
+
+  // check if there is already a valid plan fetched from OTP server, there isn't at the beginning because of async while fetching
+  useEffect(() => {
+    if (plan != {}) setValidPlan(true);
+    if (isValidPlan) {
+      console.log("PLAN:", plan);
+    }
+  }, [plan]);
+
+  
+  function ShowTravelPlan() {
+    // if you click on the map you will fly to the start coordinates of the travel plan
+    const map = useMapEvents({
+      click() {
+        if (isValidPlan) {
+          const fromLocation = { lat: plan.from.lat, lon: plan.from.lon };
+          const toLocation = { lat: plan.to.lat, lon: plan.to.lon };
+          setPositionFrom(fromLocation);
+          setPositionTo(toLocation);
+          map.flyTo(fromLocation, map.getZoom());
+        }
+      },
+    });
+
+    if (positionFrom === null) {
+      return null;
+    }
+
+    //adds markers to the start and stop coordinates of the travel plan
+    new L.marker(positionFrom, { icon: fromIcon }).addTo(map);
+    new L.marker(positionTo, { icon: toIcon }).addTo(map);
+
+    // legGeometry.points holds encoded coordinates (lat, lon) of intermediate steps along the travel plan
+    const polyUtil = require("polyline-encoded");
```

Comment:
You could probably move this to the very top of the file as an import like 
```suggestion
    import polyUtil from "polyline-encoded";
```

Reply:
done